### ISSUE TITLE
Tracking based on prev frame predictions

### ISSUE THREAD

**Issue by RaunakKalani** [tangyuhao/GOTURN-Tensorflow#17]
Hi,

I modified the code so that now it uses the predictions of the previous frame to crop the image and predict the object position in the current frame, but the tracker always predicts a bbox which is centred in the cropped frame even though the object is not there.
`The "Try" Loop

	for i in range(0, int(len(train_box)/BATCH_SIZE)):
		cur_batch = sess.run(batch_queue)
		if(i> 0):
			print "before:",cur_batch[2]
			for x in range(BATCH_SIZE):
				cur_batch[2][x][0] = 10.0*((fc4.astype(np.float64)[x][0]*prev_frames_info[x][2]/10.0)+prev_frames_info[x][0])/227.0
				cur_batch[2][x][1] = 10.0*((fc4.astype(np.float64)[x][1]*prev_frames_info[x][3]/10.0)+prev_frames_info[x][1])/227.0
				cur_batch[2][x][2] = 10.0*((fc4.astype(np.float64)[x][2]*prev_frames_info[x][2]/10.0)+prev_frames_info[x][0])/227.0
				cur_batch[2][x][3] = 10.0*((fc4.astype(np.float64)[x][3]*prev_frames_info[x][3]/10.0)+prev_frames_info[x][1])/227.0
			print "after :",cur_batch[2]
			print "\n"
			prev_frames_info = []
		for c in range(BATCH_SIZE):
			ser = cur_batch[0][c]
			tar = cur_batch[1][c]
			[x1,y1,x2,y2]  = map(int,(227*cur_batch[2][c])/10.0)
			hi = int(round(abs(y2-y1)))
			y1 = max(y1-hi/2,1)
			wi = int(round(abs(x2-x1)))   
			x1 = max(x1-wi/2,1)
			ser_resz = ser[y1:y1+2*hi+1,x1:x1+2*wi+1]
			tar_resz = tar[y1:y1+2*hi+1,x1:x1+2*wi+1]
			if(i==0):
				prev_frames_info.append([x1,y1,ser_resz.shape[1],ser_resz.shape[0]])
				height_prev = ser_resz.shape[0]
				width_prev = ser_resz.shape[1]
			else:
				prev_frames_info.append([x1,y1,width_prev,height_prev])
			cur_batch[0][c] = scipy.misc.imresize(ser_resz,(227,227))
			cur_batch[1][c] = scipy.misc.imresize(tar_resz,(227,227))
		start_time = time.time()
		[batch_loss, fc4] = sess.run([tracknet.loss, tracknet.fc4],feed_dict={tracknet.image:cur_batch[0],tracknet.target:cur_batch[1], tracknet.bbox:cur_batch[2]})
		logging.info('batch box: %s' %(fc4))
		logging.info('gt batch box: %s' %(cur_batch[2]))
		logging.info('batch loss = %f'%(batch_loss))
		logging.debug('test: time elapsed: %.3fs.'%(time.time()-start_time))
		for j in range(BATCH_SIZE):
			imge = cur_batch[0][j]
			cv2.rectangle(imge,(int(round(227*(fc4[j][0])/10)),int(round(227*fc4[j][1]/10))),(int(round(227*(fc4[j][2])/10)),int(round(227*fc4[j][3]/10))),(127,127,255),5)
			var = cur_batch[3][j].split("/")
			cv2.imwrite(PATH,imge)
except KeyboardInterrupt:
	print("get keyboard interrupt")`

Any thing that I am doing wrong. If I let it recalibrate after every alternate frame then also it fails, still predicts correct in the frame where gt bbox was provided but fails miserably in the next frame.
